plugin/documents.py

```diff
@@ -842,31 +849,37 @@ def _register_async(self) -> None:
     def _on_view_updated_async(self) -> None:
         self._code_lenses_debouncer_async.debounce(
             self._do_code_lenses_async, timeout_ms=self.code_lenses_debounce_time)
-        different, current_region = self._update_stored_region_async()
-        if not different:
+        changed_region, any_different = self._update_stored_selection_async()
+        if changed_region is None:
             return
         self._clear_highlight_regions()
         if userprefs().document_highlight_style:
             self._when_selection_remains_stable_async(
-                self._do_highlights_async, current_region, after_ms=self.highlights_debounce_time)
+                self._do_highlights_async, changed_region, after_ms=self.highlights_debounce_time)
         self.do_signature_help_async(manual=False)
 
-    def _update_stored_region_async(self) -> Tuple[bool, sublime.Region]:
+    def _update_stored_selection_async(self) -> Tuple[Optional[sublime.Region], bool]:
         """
-        Stores the current first selection in a variable.
+        Stores the current selection in a variable.
         Note that due to this function (supposedly) running in the async worker thread of ST, it can happen that the
-        view is already closed. In that case it returns Region(-1, -1). It also returns that value if there's no first
+        view is already closed. In that case it returns `None`. It also returns that value if there's no first
         selection.
 
-        :returns:   A tuple with two elements. The second element is the new region, the first element signals whether
-                    the previous region was different from the newly stored region.
+        :returns:   A tuple with two elements. The first element returns the first selection region if it has changed.
+                    The second element signals whether any selection region has changed.
         """
-        current_region = first_selection_region(self.view)
-        if current_region is not None:
-            if self._stored_region != current_region:
-                self._stored_region = current_region
-                return True, current_region
-        return False, sublime.Region(-1, -1)
+        current_first_region = None
+        any_selection_changed = False
+        selection = self.view.sel()
+        first_region = get_first_selection_region(selection)
+        if first_region is not None:
+            if not self._stored_selection or self._stored_selection[0] != first_region:
+                current_first_region = first_region
+            current_selection = list(selection)
```

Comment:
selection is already a list, so this operation seems needless

Reply:
It's not a list. It's a custom object that acts as a list but not quite

Reply:
You're right, I think I've always used it like a list, and so I falsely assumed the return type was just `List[sublime.Region]`.